File: leetcode2/1easy/Q3477.java
package Leetcode;

/*
1. 아이디어 :

 - fruits 에 있는게 baskets보다 작거나 같은때 담겼다는 의미로 0 으로 바꿔주고
   나중에 0이 아닌게 몇개있는지 세서 값으로 리턴

2. 시간복잡도 :
O( n**2 )
3. 자료구조/알고리즘 :
- / 완전탐색
 */

public class Q3477 {
    class Solution {
        public int numOfUnplacedFruits(int[] fruits, int[] baskets) {
            for(int i = 0; i<fruits.length;i++) {
                for(int j = 0; j<baskets.length; j++) {
                    if(fruits[i] <= baskets[j]) {
                        baskets[j] = 0;
                        break;
                    }
                }
            }
            int count =0;
            for(int j = 0; j<baskets.length; j++) {
                if(baskets[j] != 0) {
                    count++;
                }
            }
            return count;



        }
    }
}

File: leetcode2/1easy/강병호/Q3477.py
<<<<<<< HEAD
class Leetcode.이재훈.이재훈.Solution:
=======
class Leetcode.이재훈.Solution:
>>>>>>> ab7f46f73d8f18bcd38c5cf06ecabae64f53a424
    def numOfUnplacedFruits(self, fruits: list[int], baskets: list[int]) -> int:
        length = len(fruits)
        placed = [False] * length
        result, j = 0, 0

        while j != length:
            for i in range(0, length):
                if placed[i]:
                    continue

                if fruits[j] > baskets[i] and i == length-1: # unplace
                    break
                elif fruits[j] <= baskets[i]:
                    placed[i] = True
                    break
            
            j+=1

        for p in placed:
            if not p:
                result += 1

        return result


if __name__ == "__main__":
<<<<<<< HEAD
    sol = Leetcode.이재훈.이재훈.Solution()
=======
    sol = Leetcode.이재훈.Solution()
>>>>>>> ab7f46f73d8f18bcd38c5cf06ecabae64f53a424

    # print(sol.numOfUnplacedFruits([4, 2, 5], [3, 5, 4]))
    print(sol.numOfUnplacedFruits([8, 5], [1, 8]))